flipperdevices/flipperzero-firmware
# [FL-3330] fbt: added hooks for build & dist environments; added FW_ORIGIN_* macro for apps & SDK

# What's new

- [FL-3330] fbt: added hooks for build & dist environments
- Added FW_ORIGIN_*  define depending on FIMRWARE_ORIGIN fbt option. Used for configuring applications for different APIs provided by fw forks
- Added support for optional `scripts/fbt/fbt_hooks.py` file with build hooks
- Added new info fields to device info

# Verification 

- Check out build commands, must contain FW_ORIGIN_Official define
- Build an app with SDK from this PR, it must also contain FW_ORIGIN_Official
- Check out `device_info` cli command output

# Checklist (For Reviewer)



[FL-3330]: https://flipperzero.atlassian.net/browse/FL-3330?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
